mmv1/third_party/terraform/fwprovider/framework_provider.go.tmpl

```diff
@@ -259,16 +268,17 @@ func (p *FrameworkProvider) Configure(ctx context.Context, req provider.Configur
         return
     }
 
-    // Configuration values are now available.
-    p.LoadAndValidateFramework(ctx, &data, req.TerraformVersion, &resp.Diagnostics, p.Version)
-    if resp.Diagnostics.HasError() {
-        return
-    }
+	// Configuration values are now available.
+    // However we don't use them; for the plugin-framework implementation of the provider
+	// we take the configuration values from the SDK implementation of the provider. This avoids duplicated logic and inconsistencies in implementation.
+	// The trade off is that we don't benefit from the new type system that differentiates Null and Unknown values, which is especially useful for strings.
```

Comment:
I think it would also be helpful to reference that this trade-off has some workarounds downstream on a resource level via the field helper functions like [GetProjectFramework](https://github.com/GoogleCloudPlatform/magic-modules/blob/1e3b6fed59b9905d4ea14e55c9e76b0aaeb7a074/mmv1/third_party/terraform/fwresource/field_helpers.go#L23-L34) that prevent passing empty provider values to resource schemas.

Reply:
Thanks I'll look into updating this and work out an appropriate centralised place for this knowledge to be recorded

Reply:
I'll update the comment here, but I'm unsure where an appropriate central spot would be to record this knowledge. Please let me know if you have ideas. A Google doc maybe?